woocommerce/google-listings-and-ads
# Refactor/middleware requests

### Changes proposed in this Pull Request:

This PR refactors the Proxy class and renames it to Middleware. In most of our classes we already renamed it to Middleware using an alias.

The following steps were taken in the Proxy.php class to allow for easier testing:
- `disconnect_ads_account` is no longer used
- Ads currency functions moved to API\Google\Ads.php 
- `get_connected_ads_account` moved to Ads\AccountService.php
- `get_ads_account_ids` can be moved to API\Google\Ads.php WCS doesn't handle this request directly like the other requests in Proxy.php
- `get_merchant_ids` is not used as it was renamed to `get_merchant_accounts`
- `get_merchant_id` is no longer needed as we can get it from `options->get_merchant_id`
- Move update ads functions to Ads class
- Move update merchant functions to Merchant class
- Rename proxy class to middleware
- Pass Ads class to BudgetController since the currency functions got moved from Middleware

Unit tests were added for the classes:
- Ads
- Middleware

Closes #388

### Detailed test instructions:
The important part is to confirm that the unit tests pass and cover the functionality within each class.

In addition to the unit tests we can also test the refactored functionality through the onboarding UI, this will need to be done by testing all the steps which send a manager request to the WCS.

1. Get list of Merchant account 
2. Link an existing Merchant account
3. Create a new Merchant account
4. Link an existing Ads account
5. Create a new Ads account
6. Claim website override

Detailed steps for reproducing these can be found in #1208 and #1226
For the last step we need to verify the site after it's already been verified on another site, see instructions "Switch URL for a previously created site" in #1226 

### Changelog entry

* Update - Refactor the Middleware class and add unit testing.
